app/src/main/java/com/dessalines/thumbkey/utils/Utils.kt

```diff
@@ -747,6 +747,20 @@ fun autoCapitalizeCheck(ime: IMEService): Boolean {
     // Knows if its an empty field
     val empty = ime.currentInputConnection.getTextBeforeCursor(1, 0).isNullOrEmpty()
 
+    // Avoid capitalizing in certain edit boxes
+    val inputType = ime.currentInputEditorInfo.inputType and (InputType.TYPE_MASK_VARIATION)
+    if (listOf(
+            InputType.TYPE_TEXT_VARIATION_URI,
+            InputType.TYPE_TEXT_VARIATION_PASSWORD,
+            InputType.TYPE_TEXT_VARIATION_WEB_PASSWORD,
+            InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD,
+        ).contains(inputType)
+    ) {
+        return false
```

Comment:
Would probably be best to move this check into its own function. And this check should probably be done on L721, IE: 

`if (autoCapitalize && !isPasswordField(ime) && autoCapitalizeCheck(ime))`

Reply:
I don't really understand this, why is the password field a special case? I feel like either we should make all of them extra functions or none of them, is that not correct?

Reply:
You can call the function `specialField`, doesn't matter to me. It just should be its own function, and should be moved up to L721;

Reply:
I think you are misunderstanding me, currently, all of those input types are together in one function, that automatically says if it's one of these, then don't capitalize, right?

Why should passwords in particular be a separate function from the rest of those? Is there a technical reason that I'm blind to?

I've probably implemented this in my most recent commit,  and it works for me, but since I lack the understanding of why this is a necessary change, I may have made a mistake that testing has not revealed to me